src/DIRAC/WorkloadManagementSystem/Client/PilotLoggingPlugins/FileCacheLoggingPlugin.py

```diff
@@ -0,0 +1,125 @@
+"""
+File cache logging plugin.
+"""
+import os
+import json
+import re
+from DIRAC import S_OK, S_ERROR, gLogger
+from DIRAC.WorkloadManagementSystem.Client.PilotLoggingPlugins.PilotLoggingPlugin import PilotLoggingPlugin
+
+sLog = gLogger.getSubLogger(__name__)
+
+
+class FileCacheLoggingPlugin(PilotLoggingPlugin):
+    """
+    File cache logging. Log records are appended to a file, one for each pilot.
+    It is assumed that an agent will be installed together with this plugin, which will copy
+    the files to a safe place and clear the cache.
+    """
+
+    def __init__(self):
+        """
+        Sets the pilot log files location for a WebServer.
+
+        """
+        # UUID pattern
+        self.pattern = re.compile(r"^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$")
+        # pilot stamp pattern
+        self.stamppattern = re.compile(r"^[0-9A-F]{8}$")
```

Comment:
I changed PILOTSTAMP in https://github.com/DIRACGrid/DIRAC/pull/7059 so this should probably be adapted.